modules/home-age.nix

```
@@ -0,0 +1,233 @@
+    let
+      app = pkgs.writeShellApplication {
+        name = "agenix-home-mount-secrets";
+        runtimeInputs = with pkgs; [ coreutils ];
+        text = ''
+          ${newGeneration}
+          ${installSecrets}
+          exit 0
+        '';
+      };
+    in
+    lib.getExe app;
+
+  userDirectory = dir:
+    let
+      inherit (pkgs.stdenv.hostPlatform) isDarwin;
+      baseDir =
+        if isDarwin
+        then "$(getconf DARWIN_USER_TEMP_DIR)"
+        else "$XDG_RUNTIME_DIR";
+    in
+    "${baseDir}/${dir}";
+
+  userDirectoryDescription = dir: ''
+    "$XDG_RUNTIME_DIR"/${dir} on linux or "$(getconf DARWIN_USER_TEMP_DIR)"/${dir} on darwin.
+  '';
+in
+{
+  imports = [
+    (mkRenamedOptionModule [ "age" "sshKeyPaths" ] [ "age" "identityPaths" ])
+  ];
+
+  options.age = {
+    package = mkPackageOption pkgs "rage" { };
+
+    secrets = mkOption {
+      type = types.attrsOf secretType;
+      default = { };
+      description = ''
+        Attrset of secrets.
+      '';
+    };
+
+    identityPaths = mkOption {
+      type = types.listOf types.path;
+      default =
+        if (config.services.openssh.enable or false)
+        then map (e: e.path) (lib.filter (e: e.type == "rsa" || e.type == "ed25519") config.services.openssh.hostKeys)
+        else if isDarwin
+        then [
+          "/etc/ssh/ssh_host_ed25519_key"
+          "/etc/ssh/ssh_host_rsa_key"
+        ]
+        else [ ];
+      description = ''
+        Path to SSH keys to be used as identities in age decryption.
+      '';
+    };
+
+    secretsDir = mkOption {
+      type = types.str;
+      default = userDirectory "agenix-home";
+      defaultText = userDirectoryDescription "agenix-home";
+      description = ''
+        Folder where secrets are symlinked to
+      '';
+    };
+
+    secretsMountPoint = mkOption {
+      default = userDirectory "agenix-home.d";
+      defaultText = userDirectoryDescription "agenix-home.d";
+      description = ''
+        Where secrets are created before they are symlinked to ''${cfg.secretsDir}
+      '';
+    };
+  };
+
+  config = mkIf (cfg.secrets != { }) {
+    assertions = [
+      {
+        assertion = cfg.identityPaths != [ ];
+        message = "age.identityPaths must be set.";
+      }
+    ];
+
+    systemd.user.services.agenix = lib.mkIf pkgs.stdenv.hostPlatform.isLinux {
+      Unit = {
+        Description = "agenix activation";
+      };
+      Service = {
+        Type = "oneshot";
+        ExecStart = mountingScript;
+      };
+      Install.WantedBy = [ "default.target" ];
+    };
+
+    launchd.agents.activate-agenix = {
+      config = {
+        ProgramArguments = [ mountingScript ];
+        KeepAlive = {
+          Crashed = false;
+          SuccessfulExit = false;
+        };
+        ProcessType = "Background";
+        StandardOutPath = "${config.home.homeDirectory}/Library/Logs/AgeNix/stdout";
+        StandardErrorPath = "${config.home.homeDirectory}/Library/Logs/AgeNix/stderr";
```

```suggestion
        StandardOutPath = "${config.home.homeDirectory}/Library/Logs/agenix/stdout";
        StandardErrorPath = "${config.home.homeDirectory}/Library/Logs/agenix/stderr";
```

No fancy capitalization please.